src/Engines/CollectionEngine.php

```diff
@@ -177,20 +177,29 @@ public function columnSearch()
         for ($i = 0, $c = count($columns); $i < $c; $i++) {
             if ($this->request->isColumnSearchable($i)) {
                 $this->isFilterApplied = true;
+                $regex = $this->request->isRegex($i);
 
                 $column  = $this->getColumnName($i);
                 $keyword = $this->request->columnKeyword($i);
 
                 $this->collection = $this->collection->filter(
-                    function ($row) use ($column, $keyword) {
+                    function ($row) use ($column, $keyword, $regex) {
                         $data = $this->serialize($row);
 
                         $value = Arr::get($data, $column);
 
                         if ($this->isCaseInsensitive()) {
-                            return strpos(Str::lower($value), Str::lower($keyword)) !== false;
+                            if($regex) {
+                                return preg_match('/' . Str::lower($keyword) . '/', Str::lower($value)) == 1;
```

Comment:
I think we can use preg_match with `i` flag as per this [SO post](http://stackoverflow.com/questions/12411037/how-do-i-make-this-preg-match-case-insensitive)?